lib/rspec/mocks/message_expectation.rb

```diff
@@ -176,6 +176,7 @@ def and_throw(*args)
       def and_yield(*args, &block)
         raise_already_invoked_error_if_necessary(__method__)
         yield @eval_context = Object.new if block
+        @args_to_yield ||= []
```

Comment:
Given that it's common to refactor from this to initializing it in `initialize`, it might be worth leaving a comment explaining the reason we're not doing that here so we don't accidentally refactor back.